scrunch/datasets.py

```diff
@@ -2220,13 +2234,22 @@ def __iter__(self):
     def __len__(self):
         return len(self.resource.subvariables.index)
 
+    def variable_names(self):
+        """
+        Simply return a list of all subvariable names in the Dataset
+        """
+        sv_names = []
+        for var in self.resource.subvariables.index.items():
```

Comment:
There seems to be a lot of overlapping between DatasetVariablesMixin and DatasetSubvariablesMixin but they don't have the right attribute stored.

If we stored `self._catalog` where we send `dataset.variables` for DatasetVariablesMixin and `variable.subvariables` for DatasetSubvariablesMixin then all these `variable_names`, `iterkeys` and iterator methods wouldn't need to be re implemented as they could just go `self._catalog.by('name').keys()`, and such without needing to do one implementation for `resource.variables` and another copy for `resource.subvariables`.